synapse/federation/federation_client.py

```diff
@@ -280,15 +280,11 @@ async def backfill(
         logger.debug("backfill transaction_data=%r", transaction_data)
 
         if not isinstance(transaction_data, dict):
-            # TODO we probably want an exception type specific to federation
-            # client validation.
-            raise TypeError("Backfill transaction_data is not a dict.")
+            raise InvalidResponseError("Backfill transaction_data is not a dict.")
```

Comment:
This is mostly a drive-by, but we seem to have added a specific type for this at some point...